inst/@sym/sym.m

```diff
@@ -189,179 +188,224 @@
-        warning('possibly unintended decimal point in constructor string');
-      end
-      % x is raw sympy, could have various quotes in it
-      cmd = sprintf('z = sympy.S("%s")', strrep(x, '"', '\"'));
+    else % S() in other case
+
+      assert (isempty (asm), 'Only symbols can have assumptions.')
 
-    else % useSymbolNotS
-      if (isempty(asm))
-        cmd = sprintf('z = sympy.Symbol("%s")', x);
+      % TODO: figure version might warn on expression strings
+      %if (check)
+        % Check if the user try to execute operations from sym
+        %if (~isempty (regexp (xc, '\!|\&|\^|\:|\*|\/|\\|\+|\-|\>|\<|\=|\~')))
+        %  warning ('Please avoid execute operations from sym function.');
+        %end
+      %end
 
-      elseif (isscalar(asm) && isscalar(asm{1}) && isstruct(asm{1}))
-        % we have an assumptions dict
-        cmd = sprintf('return sympy.Symbol("%s", **_ins[0]),', x);
-        s = python_cmd (cmd, asm{1});
+      % Usually want rational output here (i.e., if input was "1.2").
+      % But if input has words and parentheses it might be raw Sympy code.
+      if (isempty (regexp (x, '\w\(.*\)')))
+        s = python_cmd (['return S("' x '", rational=True)']);
         return
+      end
 
-      elseif (iscell(asm))
-        valid_asm = assumptions('possible');
-        for n=1:length(asm)
-          assert(ischar(asm{n}), 'sym: assumption must be a string')
-          assert(ismember(asm{n}, valid_asm), ...
-                 'sym: that assumption is not supported')
-        end
-        cmd = ['z = sympy.Symbol("' x '"' ...
-               sprintf(', %s=True', asm{:}) ')'];
-      else
-        error('sym: invalid extra input, perhaps invalid assumptions?');
+      cmd = {'x = "{s}"'
+             'try:'
+             '    return (0, (0, 0), S(x))'
+             'except Exception as e:'
+             '    lis = set()'
+             '    if "(" in x or ")" in x:'
+             '        x2 = split("\(|\)| |,", x)'
+             '        x2 = [p for p in x2 if p]'
+             '        for i in x2:'
+             '            try:'
+             '                if eval("callable(" + i + ")"):'
+             '                    lis.add(i)'
+             '            except:'
+             '                pass'
+             '    if len(lis) > 0:'
+             '        return (str(e), (1, "" if len(lis) == 1 else "s"), "\", \"".join(str(e) for e in lis))'
+             '    return (str(e), (2, 0), x)' };
+
+      [err flag s] = python_cmd (strrep (cmd, '{s}', x));
+
+      switch (flag{1})
+        case 1  % Bad call to python function
+          error (['Python: %s\n' ...
+                  'Error occurred using "%s" Python function, perhaps use another variable name?'],
+                 err, s);
+        case 2  % Something else
+          error (['Python: %s\n' ...
+                  'Seems you cannot use "%s" for a variable name; perhaps this is a bug?'],
```

Comment:
Hi!, with this text i think is possible simplify the flag structure, just to flag:
```
+             '        return (str(e), (1, "" if len(lis) == 1 else "s"), "\", \"".join(str(e) for e in lis))'
 +             '    return (str(e), (2, 0), x)' };
```
to
```
+             '        return (str(e), 1, "\", \"".join(str(e) for e in lis))'
 +             '    return (str(e), 2, x)' };
```
and change ```flag{1}``` to ```flag```

Thx. Cya.